src/test/ui/pattern/slice-pattern-const-3.rs

```diff
@@ -1,24 +1,31 @@
-// compile-pass
+#![deny(unreachable_patterns)]
 
 fn main() {
     let s = &["0x00"; 4][..]; //Slice of any value
     const MAGIC_TEST: &[&str] = &["4", "5", "6", "7"]; //Const slice to pattern match with
     match s {
         MAGIC_TEST => (),
         ["0x00", "0x00", "0x00", "0x00"] => (),
-        ["4", "5", "6", "7"] => (), // this should warn
+        ["4", "5", "6", "7"] => (), // FIXME(oli-obk): this should warn, but currently does not
```

Comment:
These warn in Nightly, so this change regresses these cases. Can these be fixed in this PR?

Reply:
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=805c941847e8812026a3f27753c8a9ad does not warn on nightly and is an ICE on stable and beta. How did you test it?